src/pause/140C70.c

```diff
@@ -1,17 +1,281 @@
+        if (mapCursorCurrentOption != -1) {
+            stringWidth = get_string_width((mapCursorCurrentOption * 3) + pause_map_tab_places_description_string_id, 0);
+            stringID = (mapCursorCurrentOption * 3) + pause_map_tab_places_description_string_id;
+            xPos = arg2 + ((textOffsetX - stringWidth) >> 1);
+            yPos = textOffsetY + 1;
+            draw_string(stringID, xPos, yPos, 0xFF, 0, 0);
+        }
+    }
+}
+
+#ifdef NON_MATCHING
+void pause_map_init(s8* arg0) {
+    PauseMapSpaces* map_spaces;
+    PauseMapSpaces* map_spacesTemp;
+    Pause80250668* phi_v0;
+    f32 temp_f6;
+    s32 i = 0;
+    s32 tempEvt;
+    s32* phi_s1 = &D_8024FA30;
+    s32* hudElementPtr = &D_80270700;
+    
+    
+    for (i; i < 1; i++) {
+        hudElementPtr[i] = create_hud_element(phi_s1[0]);
+        set_hud_element_flags(hudElementPtr[i], 0x80);
+        phi_s1++;
+    }
+    
+    i = 6;
+    phi_v0 = &D_80250668;
+
+    for (i; i >= 0; i--) {
+        phi_v0->unk_10 = arg0;
+        phi_v0--;
+    }
+
+    setup_pause_menu_tab(&D_80250590, 7);
+    mapCursorCurrentOption = -1;
+    D_80270724 = 0;
+    mapCursorCurrentOptionCopy = -1;
+    tempEvt = evt_get_variable(0, 0xF5DE0329);
+    i = 0;
+    map_spaces = (PauseMapSpaces*)&D_802502B8;
+    map_spacesTemp = map_spaces;
+    
+    for (i; i < EVT_SAVE_FLAG_PLACES_VISITED_TOTAL; i++) {
+        if (map_spaces->tabIndex == tempEvt) {
+            break;
+        } else {
+            map_spaces++;
+        }
+    }
+
+    if (i < EVT_SAVE_FLAG_PLACES_VISITED_TOTAL) {
+        mapMarioXPos = map_spacesTemp[i].xPos;
+        mapMarioYPos = map_spacesTemp[i].yPos;
+    } else {
+        mapMarioXPos = 0;
+        mapMarioYPos = 0;
+    }
+    mapCursorXPos = (s32) mapMarioXPos;
+    mapCursorYPos = (s32) mapMarioYPos;
+    D_80270704 = 0.0f;
+    D_80270708 = 0;
+    temp_f6 = D_80270704;
+    
+    D_80270704 = temp_f6 - (s32) (mapCursorXPos + temp_f6 - D_80270080);
+    D_80270708 = temp_f6 - (s32) (mapCursorYPos + temp_f6 - D_80270088);
+    
+    if (0 < D_80270704) {
+        D_80270704 = 0;
+    }
 
+    if (0 < D_80270708) {
+        D_80270708 = 0;
+    }
+    
+    if (D_80270704 <= -86.0f) {
+        D_80270704 = -85.0f;
+    }
+
+    if (D_80270708 <= -210.0f) {
+        D_80270708 = -209.0f;
+    }
+    *arg0 = 1;
+}
+#else
 INCLUDE_ASM(s32, "pause/140C70", pause_map_init);
+#endif
+
+void pause_map_handle_input(void) {
+    GameStatus* gameStatus = gGameStatusPtr;
+    PauseMapSpaces* map_spaces;
+    f64 temp_f4_2;
+    s32 temp_f12;
+    f32 phi_f8;
+    f32 phi_f6;
+    f32 mapCursorYPosTemp = mapCursorYPos;
+    f32 mapCursorXPosTemp = mapCursorXPos;
+
+    phi_f8 = (f32) gameStatus->stickX * 0.05f;
+    phi_f6 = (f32) -(s32) gameStatus->stickY * 0.05f;
+    if ((phi_f8 == 0.0f) && (phi_f6 == 0.0f)) {
+        if (mapCursorCurrentOption != -1) {
+            map_spaces = (PauseMapSpaces*)&D_802502B8 + mapCursorCurrentOption;
```

Comment:
D_802502B8[mapCursorCurrentOption]

Might even be able to remove the map_spaces temp.